purchases/src/main/kotlin/com/revenuecat/purchases/listenerConversions.kt

```diff
@@ -187,15 +187,15 @@ fun Purchases.purchaseProductWith(
  * @param [onSuccess] Will be called after the purchase has completed
  * @param [onError] Will be called after the purchase has completed with error
  */
-fun Purchases.purchaseProductWithOption(
+fun Purchases.purchaseProductOptionWith(
```

Comment:
I think we will be able to drop these listeners one day since we now require Java 8 and lambda functions can be used. We don't really need the callback interfaces anymore I think

Reply:
I think ideally we would deprecate the ones with the callbacks in `Purchases.kt` and leave these versions using lambdas? Since they seem easier to use. In any case, this shouldn't block this PR.

Reply:
yes exactly, sorry wasn't clear..